src/main/java/dp/handler/Handler.java

```diff
@@ -84,9 +86,23 @@ public void listen(final ExportedFile message) {
 
         } catch (final IOException e) {
             LOGGER.error("error when exporting filter {}. exception : {}", message.getFilterId(), e);
+        } finally {
+            LOGGER.info("cleaning up workbook resources");
+            closeWorkbook((SXSSFWorkbook) workbook);
         }
+        LOGGER.info("completed processing kafka message", message.getFilterId());
+    }
 
-        LOGGER.info("exported completed for filterID: {}", message.getFilterId());
+    private void closeWorkbook(SXSSFWorkbook workbook) {
+        if (workbook != null) {
+            try {
+                // you must call dispose on SXSSFWorkbook to discard the generated temp files.
```

Comment:
does this mean it's writing the file to disk while it works on it? if we weren't doing this before it may be worth a discussion on the impact of using local storage/state

Reply:
I dont't know, I don't think so. This was something that was specifically highlighted in the [docs](http://poi.apache.org/spreadsheet/how-to.html#sxssf) for the implantation of the workbook we are now using. Basically you  ***MUST*** clean up once you are done.

Reply:
As discussed we need to have a conversation on what writing to temp files on disk means / if we can/should/are allowed to do it.

Reply:
Might be worth understanding how to configure the temporary location (a quick google suggests its possible but not sure exactly how) in case we need it.

I'm happy with these changes, just want to make sure we understand anything we might need to adjust for deployment :)